src/core/output.js

```diff
@@ -19,7 +20,10 @@ module.exports = async (entrypoints, props) => {
       const outputJson = syspath.join(item.outputDir, 'index.json')
 
       await fs.outputFile(outputHtml, template)
-      await fs.outputJson(outputJson, { content: item.content })
+      await fs.outputJson(outputJson, {
+        content: item.content,
+        toc: toc(item.content).json
```

Comment:
I'd extract this all to another function since you're using it again in `src/core/socket.js`.